### PR TITLE
[5.0.1] User profile edit

### PR DESCRIPTION
**Pull request by alessioscalici**
The User Profile edit page

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: src/helpers/entities.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+const FORM_DATE_FORMAT = 'DD/MM/YYYY';
+const API_DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';
+const API_TIMESTAMP_FORMAT = 'YYYY-MM-DD';
+
+export const zeroFill = number => padStart(number, 2, '0');
+
+export const getComponentType = (component) => {
+  switch (component) {
+    case 'Boolean':
```
Comment (fedelocci) [inline src/helpers/entities.js]: i suggest you to use the const.js in state/data-types/const.js

File: src/helpers/entities.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      case 'ThreeState': {
+        if (value === undefined) {
+          formAttr[code] = null;
+        } else {
+          formAttr[code] = value.toString();
+        }
+        break;
+      }
+      case 'Date': {
+        const newDate = moment(value, API_DATE_FORMAT).format(FORM_DATE_FORMAT);
```
Comment (fedelocci) [inline src/helpers/entities.js]: you can generalize this in a function like getDate

File: src/helpers/entities.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        formAttr[code] = newDate;
+        break;
+      }
+      case 'Timestamp': {
+        const hours = moment(value).hours();
```
Comment (fedelocci) [inline src/helpers/entities.js]: you can generalize uising an external function form example getTimestamp

File: src/helpers/entities.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        const minutes = moment(value).minutes();
+        const seconds = moment(value).seconds();
+        const newDate = moment(value, API_TIMESTAMP_FORMAT).format(FORM_DATE_FORMAT);
+        formAttr[`${code}_ts_hours`] = zeroFill(hours);
+        formAttr[`${code}_ts_minutes`] = zeroFill(minutes);
+        formAttr[`${code}_ts_seconds`] = zeroFill(seconds);
+        formAttr[code] = newDate;
+        break;
+      }
+      case 'Hypertext':
```
Comment (fedelocci) [inline src/helpers/entities.js]: like above

File: src/ui/user-profile/common/UserProfileForm.js [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+import { getComponentType } from 'helpers/entities';
+
+import FormLabel from 'ui/common/form/FormLabel';
+import RenderListField from 'ui/common/form/RenderListField';
+import { BOOLEAN_OPTIONS, THREE_STATE_OPTIONS } from 'ui/users/common/const';
+
+
+const getComponentOptions = (component) => {
+  switch (component) {
+    case 'Boolean':
```
Comment (fedelocci) [inline src/ui/user-profile/common/UserProfileForm.js]: using const.js like above

File: src/ui/user-profile/common/UserProfileForm.js [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+    { const itemsList = items.split(separator);
+      itemsList.forEach((item) => {
+        options.push({ optionDisplayName: item, value: item });
+      });
+      return options;
+    }
+    case 'EnumeratorMap': {
+      const itemsList = items.split(separator);
+      itemsList.forEach((item) => {
+        const itemKey = item.split('=')[0].trim();
```
Comment (fedelocci) [inline src/ui/user-profile/common/UserProfileForm.js]: const[ itemKEy,itemValue ] = item.split('=');

**Comment by entando-service** [comment]
All is well

**Pull request merged**